catalog/core/catalog-core-directorymonitor/src/main/java/org/codice/ddf/catalog/content/monitor/ContentDirectoryMonitor.java

```diff
@@ -316,30 +325,100 @@ private String getBlackListAsRegex() {
+        verifyCamelComponentIsAvailable("catalog");
+        camelContext.addRoutes(routeBuilder);
+      } catch (Exception e) {
+        LOGGER.info(
+            "Unable to configure Camel route. Resources will not be ingested. CDM configuration may still exist and should be remade",
+            e);
+      } finally {
+        if (LOGGER.isDebugEnabled()) {
+          dumpCamelContext("after attemptAddRoutes()");
+        }
+      }
+    }
   }
 
-  /*
-     Task that waits for the "content" and "catalog" CamelComponents before adding the routes from the
-     RouteBuilder to the CamelContext. This ensures content directory monitors will
-     automatically start after a system shutdown.
-  */
-  private void attemptAddRoutes() {
-    LOGGER.trace(
-        "Attempting to add routes for content directory monitor watching {}", monitoredDirectory);
-    try {
-      RouteBuilder routeBuilder = createRouteBuilder();
-      verifyCamelComponentIsAvailable("content");
-      verifyCamelComponentIsAvailable("catalog");
-      camelContext.addRoutes(routeBuilder);
-      setRouteCollection(routeBuilder);
-    } catch (Exception e) {
-      LOGGER.info("Unable to configure Camel route.", e);
-      INGEST_LOGGER.warn("Unable to configure Camel route.", e);
-    } finally {
-      if (LOGGER.isDebugEnabled()) {
-        dumpCamelContext("after attemptAddRoutes()");
+  /**
+   * Blocking method that waits for {@link InputTransformer} services.
+   *
+   * @throws IllegalStateException if the CDM configuration was deleted while waiting for
+   *     InputTransformers
+   * @throws InterruptedException if the thread was interrupted while waiting for InputTransformers
+   */
+  @SuppressWarnings(
+      "squid:S135" /* 2 breaks are needed in the for loop for the 2 exit conditions */)
+  private void waitForInputTransformers() throws InterruptedException {
+    synchronized (lock) {
+      final String propertyId = "id";
+      Set<String> propertyValues = new HashSet<>(inputTransformerIds.getIds());
+
+      if (propertyValues.isEmpty()) {
+        return;
+      }
+
+      long lastChecked = new Date().getTime();
+      long end = lastChecked + transformerWaitTimeoutMillis;
+      for (; ; ) {
+        if (destroyed) {
+          throw new IllegalStateException();
+        }
+
+        for (ServiceReference serviceReference : inputTransformers) {
+          Object propertyValue = serviceReference.getProperty(propertyId);
+          if (propertyValue instanceof String) {
+            propertyValues.remove(propertyValue);
+            LOGGER.debug("Remaining property values {}", propertyValues);
+          }
+        }
+
+        if (propertyValues.isEmpty()) {
+          LOGGER.trace("Finished finding all InputTransformers");
+          break;
+        }
+
+        if (lastChecked >= end) {
+          LOGGER.warn(
+              "Not all input transformers were found, starting up CDM route anyways. Ingested resources may not have expected metadata");
```

Comment:
❓ What should an administrator do about this situation? Could we log the missing input transformers?

Reply:
I ended up switching this to fail if not all InputTransformers are not found and beefed up the message with potential solutions.